### PR TITLE
Update client-generator

### PR DESCRIPTION
**Pull request by wojtek-t**

### CODE REVIEW

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this please

File: cmd/libs/go2idl/client-gen/generators/generator-for-type.go [hunk 1]
```diff
@@ -157,6 +159,31 @@ func (c *$.type|privatePlural$) Delete(name string, options *$.apiDeleteOptions|
 }
 `
 
+var deleteCollectionTemplate = `
+// DeleteCollection deletes a collection of objects.
+func (c *$.type|privatePlural$) DeleteCollection(options *$.apiDeleteOptions|raw$, listOptions $.apiListOptions|raw$) error {
+	if options == nil {
+		return e.Client.Delete().
```
Comment (caesarxuchao) [inline cmd/libs/go2idl/client-gen/generators/generator-for-type.go]: Should be `c.Client...`, not `e` :)

File: cmd/libs/go2idl/client-gen/generators/generator-for-type.go [hunk 2]
```diff
@@ -157,6 +159,31 @@ func (c *$.type|privatePlural$) Delete(name string, options *$.apiDeleteOptions|
...
+			Resource("$.type|privatePlural$").
+			VersionedParams(&listOptions, api.Scheme).
+			Do().
+			Error()
+	}
+	body, err := api.Scheme.EncodeToVersion(options, c.client.APIVersion())
+	if err != nil {
+		return err
+	}
+	return e.Client.Delete().
```
Comment (caesarxuchao) [inline cmd/libs/go2idl/client-gen/generators/generator-for-type.go]: same here.

**Comment by caesarxuchao** [comment]
A minor problem, otherwise lgtm. Feel free to apply the label after addressing the comment. Thank you.

ref #18290

File: cmd/libs/go2idl/client-gen/generators/generator-for-type.go (hunk 1 above)
Comment (wojtek-t) [inline cmd/libs/go2idl/client-gen/generators/generator-for-type.go]: done

File: cmd/libs/go2idl/client-gen/generators/generator-for-type.go (hunk 2 above)
Comment (wojtek-t) [inline cmd/libs/go2idl/client-gen/generators/generator-for-type.go]: done

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Comment by caesarxuchao** [comment]
LGTM. Thanks.

**Pull request merged**